geniza/common/utils.py

```diff
@@ -89,3 +91,35 @@ def __enter__(self, *x, **y):
 
     def __exit__(self, *x, **y):
         return
+
+
+class Timer:
```

Comment:
I see the potential value of this utility code, but if you write code you need to test & document it too

Reply:
was looking to see if there was an existing tool for this we could use (maybe as a dev dependency but not prod), and saw that there is a [context manager decorator]( https://docs.python.org/3.9/library/contextlib.html#contextlib.contextmanager). Maybe you already were aware and it doesn't work in this case, since you want to optionally customize print behavior, but thought I would share in case note. Looks like they added several variants in v3.7.

Reply:
Didn't know about that decorator actually but yeah it doesn't seem like it would work in this context since we want something can store the variable (it's not a try/catch/finally pattern).

Reply:
Let me see if I can find some good timer code replacement otherwise I can write a few tests, a simple timer seems useful for things like this

Reply:
We could try this?
https://pypi.org/project/contexttimer/
Not sure if any of this is worth it -- happy to just get rid of all timery stuff for now

Reply:
Yeah, I did a quick look and found that one too — looks like it hasn't been updated in a while, but if it works we could use it. 

Otherwise, maybe you can just write one test that covers all of this and add some comments or docstrings.

Or another option would be to customize the rich progressbar so it will report total time when completed.  I think right now it is only counting down, would be nice to have count down + total time. Is there anyway to start the progressbar sooner so that it will include the queries and gathering data in the total? (Or does it already?)

I like the impulse, though — I have timer + logging code that I just refer to and copy instead of refactoring into something reusable. Would be nice